### PR TITLE
Added functionality to move out of ArmedAway

### PR DESCRIPTION
**Pull request by carterclark**
### What does this do?
This adds functionality to go from: 
- ready to countdownAway
- countdownToAway to ArmedAway
- countdownToAway to countdownToNotReady
- countdownToNotReady to notReady

Also added functions to Countdown
### Why do we need it?
Progress functionality between states
### Have you included tests for your changes?
No, I don't know how to test these yet. Will use the UI to test it in the next PR

### CODE REVIEW

File: src/states/away/CountdownToAway.java [hunk 1]
```diff
@@ -1,7 +1,69 @@
...
+	public void enter() {
+
+		SecurityContext.instance().showSecondsToAway(timer.getTimeValue());
+	}
+
+	@Override
+	public void leave() {
+
+		super.leave();
+		SecurityContext.instance().showAway();
```
Comment (paveldanek) [inline src/states/away/CountdownToAway.java]: I think it should showAway, as it enters Away, not as it leaves countdown.

**Review by paveldanek — CHANGES_REQUESTED** [review]
I left a comment at CountdownToAway.

File: src/states/away/CountdownToAway.java (hunk 1 above)
Comment (carterclark) [reply src/states/away/CountdownToAway.java]: Very true, I will adapt it

File: src/states/away/CountdownToAwayNotReady.java [hunk 2]
```diff
@@ -1,7 +1,63 @@
...
+		timer = new Timer(this, 10);
+		SecurityContext.instance().showSecondsToAway(timer.getTimeValue());
+
+	}
+
+	@Override
+	public void leave() {
+		super.leave();
+		SecurityContext.instance().showNotReady();
+	}
```
Comment (paveldanek) [inline src/states/away/CountdownToAwayNotReady.java:69]: I'm sorry I didn't notice that earlier, but this is the same thing as before. Showing a new state should be the new state itself, upon entering.

File: src/states/away/CountdownToAwayNotReady.java [hunk 3]
```diff
@@ -1,7 +1,63 @@
...
+	 */
+	@Override
+	public void handleEvent(CheckAllZones event) {
+
+		SecurityContext.instance().changeState(CountdownToAway.instance());
+	}
+
+	@Override
+	public void enter() {
+		timer = new Timer(this, 10);
```
Comment (paveldanek) [inline src/states/away/CountdownToAwayNotReady.java]: Have you tried and run this code? I have a feeling, that between your states CountdownToAway and CountdownToAwayNotReady, whenever you CheckAllZones or UncheckZone, a new 10-second timer always starts... Try and run it on your computer.

**Review by paveldanek — COMMENTED** [review]
I'm sorry I didn't notice these earlier, but I left you 2 more comments. The one regarding timer is more important, though.

File: src/states/away/CountdownToAwayNotReady.java (hunk 3 above)
Comment (carterclark) [reply src/states/away/CountdownToAwayNotReady.java]: Yes, this problem is fixed in another branch. It is ready for PR whenever this one gets approved

**Review by paveldanek — APPROVED** [review]

**Pull request merged**